### PR TITLE
Container API Test with HostName

### PR DESCRIPTION
**Pull request by rajdeepd**
New Test for Creating a Container using API with custom Hostname

Signed-off-by: Rajdeep Dua dua_rajdeep@yahoo.com

### CODE REVIEW

**Comment by rajdeepd** [comment]
Got it closed by mistake hence sending a new one

File: integration-cli/docker_api_containers_test.go [hunk 1]
```diff
@@ -681,6 +681,49 @@ func (s *DockerSuite) TestContainerApiCreate(c *check.C) {
...
+	}
+	var container createResp
+	if err := json.Unmarshal(b, &container); err != nil {
+		c.Fatal(err)
+	}
+
+	var id = container.Id
+	configGet := map[string]interface{}{}
+
+	_, bodyGet, err := sockRequest("GET", "/containers/"+id+"/json", configGet)
```
Comment (duglin) [inline integration-cli/docker_api_containers_test.go]: I think you can pass in "nil" instead of an empty configGet.

**Comment by rajdeepd** [comment]
changed as suggested @duglin

File: integration-cli/docker_api_containers_test.go [hunk 2]
```diff
@@ -681,6 +681,48 @@ func (s *DockerSuite) TestContainerApiCreate(c *check.C) {
 	}
 }
 
+func (s *DockerSuite) TestContainerApiCreateWithHostName(c *check.C) {
+	defer deleteAllContainers()
```
Comment (runcom) [inline integration-cli/docker_api_containers_test.go]: @duglin do we still need this with new gocheck suite?

Comment (runcom) [inline integration-cli/docker_api_containers_test.go]: seems we don't https://github.com/rajdeepd/docker/blob/dry-run-test/integration-cli/check_test.go#L30

Comment (duglin) [inline integration-cli/docker_api_containers_test.go]: yup

**Comment by rajdeepd** [comment]
Shall i delete this line?

**Comment by duglin** [comment]
@rajdeepd yup

**Comment by rajdeepd** [comment]
deleted

**Comment by duglin** [comment]
LGTM
ping @LK4D4

**Comment by jfrazelle** [comment]
LGTM

**Pull request merged**